frontend/src/utils/fieldValidation.ts

```diff
@@ -219,6 +219,11 @@ export const createDecimalValidationRules: ValidationRuleFn<
         return 'Please enter a valid decimal'
       }
 
+      // Validate leading zeros
+      if (val.match(/^0[0-9].*$/)) {
```

Comment:
So, small correction was needed after all 😅 

```suggestion
      if (/^0+([1-9]|0$)/.test(val)) {
```

Reasoning:
1) (_functional bug_): testing the front for `/^0[0-9]/` would allow a double `0` at the lead and accept entries like `001` (oops! 😅). We have to be able to reject any number of leading zeros followed either by any digit 1 to 9 and whatever else or a zero
2) (_optimization_) `String.match()` return a data structure of the matches. For boolean-situations, better to use `RegExp.test()` instead
3) (_micro-optimization_) ending a test regex pattern with `.*$` is useless, because it simply means: "anything till the end". In such cases, better limit the test pattern to only the front part, to clearly indicate the intent of the test

 test cases
```
/^0+([1-9]|0$)/.test('000') // true
/^0+([1-9]|0$)/.test('00') // true
/^0+([1-9]|0$)/.test('0') // false

/^0+([1-9]|0$)/.test('001') // true
/^0+([1-9]|0$)/.test('01') // true
/^0+([1-9]|0$)/.test('1') // false

/^0+([1-9]|0$)/.test('0010') // true
/^0+([1-9]|0$)/.test('010') // true
/^0+([1-9]|0$)/.test('10') // false
```

Reply:
something seems off with your regex too, it will wait for decimals with leading zeros

```ts
/^0+([1-9]|0$)/.test('000000.0001') // false
```

Reply:
fixed in c87dad4